### DISCUSSION TITLE
Implementation of Shampoo

### DISCUSSION THREAD

**Discussion by shyams2** [google-deepmind/optax#285] [Q&A] [answered]
Hi

Is there an implementation of Shampoo (Anil et al, 2020) is available on optax? 
I did see that there was a PR for [this](https://github.com/deepmind/optax/pull/65). However, it seem to have been closed without merging.

**Comment by mkunesch** [comment, marked as answer]
Hi!

Thanks for the question! The code for this is ready in principle, but we'd like to test it more (for both accuracy and performance) and this is taking slightly longer. I'm currently checking how long this might still take and whether it would make sense to provide it as part of `experimental` in the meantime. 

I'll let you know as soon as I know more!

**Reply by shyams2** [reply]
I'm not in any big hurry but I did want to try it out for one of my problems. It'll be great to have it as an `experimental` feature for now.

Thanks!

**Comment by pierrot-lc** [comment]
Hi! Any news about it? The original PR is closed and I didn't see any other PR or discussion about it.
